client/allocrunnerv2/alloc_runner.go

```diff
@@ -218,6 +233,159 @@ func (ar *allocRunner) Restore() error {
 	})
 }
 
+// TaskStateUpdated is called when a task's state has been updated. This hook is
+// used to compute changes to the alloc's ClientStatus and to update the server
+// with the new state.
+func (ar *allocRunner) TaskStateUpdated(taskName string, state *structs.TaskState) error {
+	if state == nil {
+		return fmt.Errorf("nil state given to TaskStateUpdated")
```

Comment:
Think we're going to need this long term? Seems like a pretty serious programming error if we ever hit it.

Reply:
It's also the only error we return from the updater... doesn't seem worth it.